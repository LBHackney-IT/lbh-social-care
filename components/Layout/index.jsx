import AdminNavBar from 'components/AdminNavBar/AdminNavBar';
import Footer from './Footer/Footer';
import Header from './Header/Header';
import SkipLink from './SkipLink/SkipLink';
import PhaseBanner from './PhaseBanner/PhaseBanner';
import { StateProvider } from '../../utils/store';

const Layout = ({ children }) => {
  const initialState = {
    data: {},
  };

  const reducer = (state, action) => {
    switch (action.type) {
      case 'updateData':
        return {
          ...state,
          data: action.updateData,
        };
      default:
        return state;
    }
  };
  return (
    <>
      <StateProvider initialState={initialState} reducer={reducer}>
        <SkipLink />
        <Header serviceName="Social Care Service" />
        <div className="govuk-width-container app-width-container">
          <PhaseBanner phase="beta" />
          <main
            className="govuk-main-wrapper app-main-class"
            id="content"
            role="main"
          >
            <AdminNavBar />
            {children}
          </main>
        </div>
        <Footer />
      </StateProvider>
    </>
  );
};

export default Layout;
