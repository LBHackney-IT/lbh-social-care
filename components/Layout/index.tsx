import Seo from './Seo/Seo';
import Header from './Header/Header';
import SkipLink from './SkipLink/SkipLink';
import PhaseBanner from './PhaseBanner/PhaseBanner';
import BackButton from './BackButton/BackButton';
import Footer from './Footer/Footer';

export interface Props {
  children: React.ReactChild;
  goBackButton?: boolean;
}

const Layout = ({ goBackButton, children }: Props): React.ReactElement => {
  const feedbackLink = process.env.NEXT_PUBLIC_FEEDBACK_LINK || '';
  return (
    <>
      <Seo title="Social Care Admin - Hackney Council" />
      <SkipLink />
      <Header serviceName="Social Care" />
      <PhaseBanner phase="beta" feedbackLink={feedbackLink} />
      {goBackButton && <BackButton />}

      <div className="govuk-width-container">
        <main className="govuk-main-wrapper" id="content" role="main">
          {children}
        </main>
      </div>

      <Footer />
    </>
  );
};

export default Layout;
