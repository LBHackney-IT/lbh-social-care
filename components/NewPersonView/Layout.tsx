import Head from 'next/head';
import Link from 'next/link';
import { format } from 'date-fns';
import { useRouter } from 'next/router';
import { AllocationData, Resident } from 'types';
import s from './index.module.scss';
import { useRelationships } from 'utils/api/relationships';
import { useAllocatedWorkers } from 'utils/api/allocatedWorkers';
import React from 'react';
import WarningNotes from 'components/WarningNote/WarningNotes';

interface NavLinkProps {
  href: string;
  children: React.ReactChild;
}

const NavLink = ({ href, children }: NavLinkProps) => {
  const router = useRouter();
  return (
    <li>
      <Link href={href}>
        <a
          className={`lbh-link lbh-link--no-visited-state ${
            router.asPath === href && `govuk-!-font-weight-bold`
          }`}
        >
          {children}
        </a>
      </Link>
    </li>
  );
};

interface Props {
  person: Resident;
  children: React.ReactChild;
}

const summariseAllocations = (allocations): string | null => {
  if (allocations?.length === 1)
    return ` · Allocated to ${allocations[0].allocatedWorker}`;
  if (allocations?.length > 1)
    return ` · Allocated to ${allocations[0].allocatedWorker} and ${
      allocations?.length - 1
    } others`;
  return null;
};

const Layout = ({ person, children }: Props): React.ReactElement => {
  const { data: allocations, error: allocationError } = useAllocatedWorkers(
    person.id
  );
  const { data: relationships, error: relationshipsError } = useRelationships(
    person.id
  );

  const navigation = [
    { text: 'Timeline', href: `/people/${person.id}` },
    {
      text: `Allocations ${
        allocations?.allocations ? `(${allocations?.allocations?.length})` : ''
      }`,
      href: `/people/${person.id}/allocations`,
    },
    {
      text: `Relationships ${
        relationships?.personalRelationships
          ? `(${relationships?.personalRelationships?.length})`
          : ''
      }`,
      href: `/people/${person.id}/relationships`,
    },
    { text: 'Details', href: `/people/${person.id}/details` },
  ];

  const secondaryNavigation = [
    {
      text: 'Add warning note',
      href: `/people/${person.id}/warning-notes/add?id=${person.id}`,
    },
  ];

  return (
    <>
      <Head>
        <title>
          {person.firstName} {person.lastName} | Social care | Hackney Council
        </title>
      </Head>

      <WarningNotes id={person.id} />

      <div
        className={`govuk-grid-row govuk-!-margin-bottom-8 ${s.personHeader}`}
      >
        <div className="govuk-grid-column-two-thirds">
          <h1 className="lbh-heading-h1">
            {person.firstName} {person.lastName}
          </h1>
          <p
            className={`govuk-caption-m govuk-!-margin-top-3 ${s.personCaption}`}
          >
            #{person.id}
            {person.dateOfBirth &&
              ` · Born ${format(new Date(person.dateOfBirth), 'dd MMM yyyy')}`}
            {allocations?.allocations &&
              summariseAllocations(allocations.allocations)}
          </p>
        </div>

        <div className={`govuk-grid-column-one-third ${s.actionsArea}`}>
          <Link href={`/people/${person.id}/records`}>
            <a className={`govuk-button lbh-button ${s.primaryAction}`}>
              <svg width="12" height="12" viewBox="0 0 12 12">
                <path d="M6.94 0L5 0V12H6.94V0Z" />
                <path d="M12 5H0V7H12V5Z" />
              </svg>
              Add something new
            </a>
          </Link>
        </div>
      </div>

      <div className={`govuk-grid-row ${s.outer}`}>
        <div className="govuk-grid-column-one-quarter">
          <nav className={s.sticky}>
            <ul className="lbh-list">
              {navigation.map((link) => (
                <NavLink href={link.href} key={link.href}>
                  {link.text}
                </NavLink>
              ))}
            </ul>

            <ul className={`lbh-list ${s.secondaryNav}`}>
              {secondaryNavigation.map((link) => (
                <li key={link.href}>
                  <a
                    href={link.href}
                    className="lbh-link lbh-body-s lbh-link--no-visited-state"
                  >
                    {link.text}
                  </a>
                </li>
              ))}
            </ul>
          </nav>
        </div>
        <div className="govuk-grid-column-three-quarters">{children}</div>
      </div>
    </>
  );
};

export default Layout;
