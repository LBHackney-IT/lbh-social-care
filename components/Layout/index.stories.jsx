import Layout from './index';
import { RouterContext } from 'next/dist/next-server/lib/router-context';
import UserContext from 'components/UserContext/UserContext';

export default {
  title: 'Layout',
  component: Layout,
};

const Template = (args) => (
  <RouterContext.Provider
    value={{
      asPath: 'foo',
    }}
  >
    <UserContext.Provider
      value={{
        user: { name: 'bar' },
      }}
    >
      <Layout {...args} />
    </UserContext.Provider>
  </RouterContext.Provider>
);

export const Default = Template.bind({});
Default.args = {
  label: 'Layout',
};
