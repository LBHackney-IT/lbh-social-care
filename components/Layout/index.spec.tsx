import { render } from '@testing-library/react';

import Layout from './index';

jest.mock('next/router', () => ({
  useRouter: () => ({
    asPath: 'path',
  }),
}));

jest.mock('./Header/Header', () => () => 'Header');
jest.mock('./PhaseBanner/PhaseBanner', () => () => 'MockedPhaseBanner');

describe('Layout component', () => {
  it('should render properly', async () => {
    const { asFragment } = render(
      <Layout>
        <p>I am the children</p>
      </Layout>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
