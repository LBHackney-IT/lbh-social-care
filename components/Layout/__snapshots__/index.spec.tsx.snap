// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should render properly 1`] = `
<DocumentFragment>
  <a
    class="govuk-skip-link"
    href="#content"
  >
    Skip to main content
  </a>
  HeaderMockedPhaseBanner
  <div
    class="govuk-width-container"
  >
    <main
      class="govuk-main-wrapper"
      id="content"
      role="main"
    >
      <p>
        I am the children
      </p>
    </main>
  </div>
  <footer
    class="footer"
    role="contentinfo"
  >
    <div
      class="govuk-width-container"
    >
      <nav
        class="links"
      >
        <a
          href="https://sites.google.com/hackney.gov.uk/moderntoolsforsocialcare/home"
        >
          Roadmap
        </a>
        <a
          href="https://docs.google.com/forms/d/e/1FAIpQLScILbPD1ioKHzp1D3HN4_DKaxV2tpWLMu8upSSqNgSPCo85cg/viewform"
        >
          Give feedback
        </a>
      </nav>
      <div
        class="meta"
      >
        Built and maintained by HackIT.
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
